world/petitions/models.py

```diff
@@ -167,15 +173,41 @@ def display(self):
         """Gets string display of the amount purchased and by whom"""
         return "{} bought {}".format(self.buyer, self.amount)
 
+class PetitionSettings(SharedMemoryModel):
+    owner = models.ForeignKey("dominion.PlayerOrNpc", related_name="petition_settings")
+    inform=models.BooleanField(default=True)
+    ignore_general=models.BooleanField(default=False)
+    ignored_organizations=models.ManyToManyField(Organization)
+    
+    def cleanup(self):
+        self.ignore_general=False
+        self.inform=True
+        self.ignored_organizations.clear()
+        try:
+            participations=self.owner.petitionparticipation_set.all()
+            for petition_participation in participations:
+                petition_participation.subscribed=False
+                petition_participation.unread_posts=True
+                petition_participation.signed_up=False
+                if (petition_participation.is_owner):
+                    petition_participation.petition.closed=True
+                petition_participation.save()
+        except:
```

Comment:
I don't really see a need to have this in an exception block at all. It doesn't look like any exceptions are expected to be raised.